src/fragmentarium/domain/text.ts

```diff
@@ -140,21 +140,147 @@ export type Token =
   | Enclosure
   | Tabulation
 
-export type Line = RulingDollarLine | LineBase
+export type Line =
+  | TextLine
+  | EmptyLine
+  | LooseDollarLine
+  | ImageDollarLine
+  | RulingDollarLine
+  | SealDollarLine
+  | StateDollarLine
+  | SealAtLine
+  | HeadingAtLine
+  | ColumnAtLine
+  | DiscourseAtLine
+  | SurfaceAtLine
+  | ObjectAtLine
+  | DivisionAtLine
+  | CompositeAtLine
 
 export interface LineBase {
   readonly type: string
   readonly prefix: string
   readonly content: ReadonlyArray<Token>
 }
 
+export interface EmptyLine extends LineBase {
+  readonly type: 'EmptyLine'
```

Comment:
The prefix should be `""` and content `[]`.

Reply:
typescripts complains when i put `content: []` that some `.maps` and `.reduce` are not callable.

Reply:
Does it work if you add type assertion? See: https://stackoverflow.com/a/43230812

Reply:
didn't work. Type assertion doesnt seem to have same effect on interface or i missed sth.

Reply:
Then we have to leave it out.